resources/axon.rb

```diff
@@ -98,82 +99,132 @@
+    # In case we have not set the driver and service, we can assume they
+    # were manually specified and need to download them to the node.
+    node.run_state['local_eg_driver'] = ::Chef::Config['file_cache_path'] + '/eg_driver' + ext
+    node.run_state['local_eg_service'] = ::Chef::Config['file_cache_path'] + '/eg_service' + ext
 
     # Set the correct header for the eg driver installer
-    eg_driver_source = if new_resource.eg_driver_installer.start_with?('http') && new_resource.eg_install
-                         new_resource.eg_driver_installer
-                       else
-                         'file:///' + new_resource.eg_driver_installer
-                       end
+    eg_driver_source = new_resource.eg_driver_installer.start_with?('http') && new_resource.eg_install ? new_resource.eg_driver_installer : 'file:///' + new_resource.eg_driver_installer
 
     # Set the corret header for the eg service installer
-    eg_service_source = if new_resource.eg_service_installer.start_with?('http') && new_resource.eg_install
-                          new_resource.eg_service_installer
-                        else
-                          'file:///' + new_resource.eg_service_installer
-                        end
+    eg_service_source = new_resource.eg_service_installer.start_with?('http') && new_resource.eg_install ? new_resource.eg_service_installer : 'file:///' + new_resource.eg_service_installer
 
     # Install package eg if enabled & not windows
-    remote_file local_eg_driver do
+    remote_file lazy { node.run_state['local_eg_driver'] } do
       source eg_driver_source
       mode '0744'
     end
-    remote_file local_eg_service do
+    remote_file lazy { node.run_state['local_eg_service'] } do
       source eg_service_source
       mode '0744'
     end
+  end
 
-    # Install EG driver and service
-    [local_eg_driver, local_eg_service].each do |pkg|
+  # If the service installer or the driver installer are not specified, we will not install unless
+  # the agent source is a tgz file. In this case the installers do not need to be set since they were derived
+  # in the tgz.
+  eg_specified = (!new_resource.eg_driver_installer.nil? && !new_resource.eg_service_installer.nil?) || agent_source_is_tgz
+  install_eg_components = !platform?('windows') && eg_specified && new_resource.eg_install
+
+  if install_eg_components
+    # Install EG driver
+    %w(local_eg_driver local_eg_service).each do |name|
       if platform_family?('debian')
-        dpkg_package pkg do
+        dpkg_package 'pkg' do
+          package_name lazy { node.run_state[name] }
           action :install
         end
       elsif platform_family?('rhel')
-        rpm_package pkg do
+        rpm_package 'pkg' do
+          package_name lazy { node.run_state[name] }
           action :install
         end
       else
         # All other platforms we use generic packages
-        package pkg do
+        package 'pkg' do
```

Comment:
I think the reason why it was split out is that the generic pkg installer would try to use yum in some instances instead of RPM. By enforcing using the RPM package we ensure that the installer is being installed exactly as in the instructions.